File: pdu/python/sensor_msgs/pdu_cdr_conv_RelativeHumidity.py
from .pdu_pytype_RelativeHumidity import RelativeHumidity
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *



def py_to_cdr_body_RelativeHumidity(writer: CdrWriter, src: RelativeHumidity):
    py_to_cdr_body_Header(writer, src.header)
    writer.write_float64(src.relative_humidity)
    writer.write_float64(src.variance)


def cdr_body_to_py_RelativeHumidity(reader: CdrReader, dst: RelativeHumidity):
    cdr_body_to_py_Header(reader, dst.header)
    dst.relative_humidity = reader.read_float64()
    dst.variance = reader.read_float64()
    return dst


def py_to_cdr_RelativeHumidity(src: RelativeHumidity) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_RelativeHumidity(writer, src)
    return writer.bytes()


def cdr_to_py_RelativeHumidity(cdr_payload) -> RelativeHumidity:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = RelativeHumidity()
    return cdr_body_to_py_RelativeHumidity(reader, dst)

File: pdu/javascript/sensor_msgs/pdu_cdr_conv_RelativeHumidity.js
import { RelativeHumidity } from './pdu_jstype_RelativeHumidity.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';


export class PduRelativeHumidityConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { RelativeHumidity } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        writer.write_float64(src.relative_humidity);
        writer.write_float64(src.variance);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { RelativeHumidity } dst
     * @returns { RelativeHumidity }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        dst.relative_humidity = reader.read_float64();
        dst.variance = reader.read_float64();
        return dst;
    }

    /**
     * @param { RelativeHumidity } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { RelativeHumidity }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new RelativeHumidity());
    }
}
